src/main/java/com/avispl/symphony/dal/avdevices/audionetworkinterface/qsc/undnemo/QSCUndnemoCommunicator.java

```diff
@@ -0,0 +1,788 @@
+	 * to save the number of requests made to the device.
+	 *
+	 * @param value value of the new index
+	 * @param currentActiveChannelIndex current active channel index
+	 * @throws Exception if fail to get UDP response
+	 */
+	private void handleActiveChannelControl(String value, String currentActiveChannelIndex) throws Exception {
+		String response = getUDPResponse(QSCUndnemoUDPCommand.SET_ACT_CH_IDX.getCommand() + QSCUndnemoConstant.SPACE + value);
+		if (QSCUndnemoConstant.NACK.equals(response)) {
+			throw new CommandFailureException(this.getAddress(), QSCUndnemoUDPCommand.SET_ACT_CH_IDX.getCommand(), String.format("Fail to set active channel index with value: %s", value));
+		}
+		// From this line to the end of the method is used to switch places between two groups of stat/control "old active channel" and "new active channel" -
+		// This is because we want to prevent unnecessary fetch all 64 channels.
+		Map<String, String> stats = localExtendedStatistics.getStatistics();
+		String newActiveIndexKey = QSCUndnemoConstant.EMPTY;
+		if (Integer.parseInt(value) <= 9) {
+			newActiveIndexKey += String.format("Channel 0%s", value);
+		} else {
+			newActiveIndexKey += String.format("Channel %s", value);
+		}
+		String oldActiveIndexKey = QSCUndnemoConstant.ACTIVE_CHANNEL;
+		String oldActiveIndexEnableState = stats.get(String.format("%s#%s", oldActiveIndexKey, QSCUndnemoMetric.CHANNEL_INFO_ENABLE_STATE.getName()));
+		String oldActiveIndexDeviceName = stats.get(String.format("%s#%s", oldActiveIndexKey, QSCUndnemoMetric.CHANNEL_INFO_DEVICE_NAME.getName()));
+		String oldActiveIndexChannelName = stats.get(String.format("%s#%s", oldActiveIndexKey, QSCUndnemoMetric.CHANNEL_INFO_CHANNEL_NAME.getName()));
+		String oldActiveIndexDisplayName = stats.get(String.format("%s#%s", oldActiveIndexKey, QSCUndnemoMetric.CHANNEL_INFO_DISPLAY_NAME.getName()));
+		String oldActiveIndexGroupName;
+		if (Integer.parseInt(currentActiveChannelIndex) <= 9) {
+			oldActiveIndexGroupName = String.format("Channel 0%s", currentActiveChannelIndex);
+		} else {
+			oldActiveIndexGroupName = String.format("Channel %s", currentActiveChannelIndex);
+		}
+
+		String newActiveIndexEnableState = stats.get(String.format("%s#%s", newActiveIndexKey, QSCUndnemoMetric.CHANNEL_INFO_ENABLE_STATE.getName()));
+		String newActiveIndexDeviceName = stats.get(String.format("%s#%s", newActiveIndexKey, QSCUndnemoMetric.CHANNEL_INFO_DEVICE_NAME.getName()));
+		String newActiveIndexChannelName = stats.get(String.format("%s#%s", newActiveIndexKey, QSCUndnemoMetric.CHANNEL_INFO_CHANNEL_NAME.getName()));
+		String newActiveIndexDisplayName = stats.get(String.format("%s#%s", newActiveIndexKey, QSCUndnemoMetric.CHANNEL_INFO_DISPLAY_NAME.getName()));
+
+		String groupName = QSCUndnemoConstant.ACTIVE_CHANNEL;
+		// Remove previous non-active group.
+		stats.remove(String.format("%s#%s", newActiveIndexKey, QSCUndnemoMetric.CHANNEL_INFO_ENABLE_STATE.getName()));
```

Comment:
The name to remove is being calculated multiple times (here and line 381). Can be extracted into variable and done once

Reply:
Updated.